hoymiles_modbus/client.py

```diff
@@ -43,9 +41,10 @@ def _data_length_fixer(packet):  # pragma: no cover
             fixed_packet[8] = len(fixed_packet[9:])
         return bytes(fixed_packet)
 
-    def processIncomingPacket(self, data, callback, unit, **kwargs):
+    def decode(self, data: bytes) -> tuple[int, int, int, bytes]:
```

Comment:
I would expect `decode_data` instead of `decode`

Reply:
Some DTUs (at least mine) send incorrectly formatted modbus responses (incorrect value in the length field). The custom framer is to fix (recalculate length) the responses before processing them further. Without that pymodbus cannot decode responses.
Yes, probably worth commenting.